include/deal.II/lac/petsc_block_vector.h

```diff
@@ -150,6 +154,13 @@ namespace PETScWrappers
       BlockVector &
       operator=(const BlockVector &V);
 
+      /**
+       * This method assignes the PETSc Vec to the instance of the class.
+       *
+       */
+      void
+      assign_petsc_vector(Vec v);
```

Comment:
Would it be possible to use a constructor instead of this function?

Reply:
The point here is that you want to keep the same deal.ii object but associate a different PETSc `Vec`

Reply:
Ps: the version with the constructor is already there (if not in this PR, in a subsequent PR)

Reply:
I'd rather use standard C++ idioms here instead of a new member function. It may be wordier but it will make this consistent with the rest of the library (e.g., if we ever did something similar for direct manipulation of Trilinos objects).

In particular, I'd prefer to create a new object and then swap:
```cpp
PETScWrappers::Vector v1;
// set up v1

Vec v;
// set up v
PETScWrappers::Vector v2(v);
swap(v1, v2);
```
that way we don't have two functions which do the same thing (the ctor and the assign function). Are you two OK with doing it this way instead?

Reply:
I'm ok with using `swap` but I have to think about the implications and the proper implementation. Can we postpone this change? I will open an MR later for this

Reply:
@drwells Ping. Can we merge this in and continue integrating the PETSc additions? I agree with your comments and remove the `assign` method, replacing with `swap` in another PR. @luca-heltai

Reply:
https://github.com/dealii/dealii/issues/14656

Reply:
@bangerth I have renamed with`reinit` here https://github.com/dealii/dealii/pull/14481/commits/757f08692cd4996f064546764b66c6d11c31ecc0